### ISSUE TITLE
[openssl] openssl/1.0.2t: ConanException: Cannot recognize the Windows subsystem

### ISSUE THREAD

**Issue by Talkless** [conan-io/conan-center-index#175]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **openssl/1.0.2t**
  * Operating System+version: **Windows 10 amd64**
  * Compiler+version: **Visual Studio 2015 (14.0)**
  * Conan version: **conan 1.19.0**
  * Python version: **Python 3.7.4**

After upgrading `OpenSSL/1.0.2s@conan/stable` to `openssl/1.0.2t@` (as `poco` requires `openssl` not `OpenSSL`, etc), I can't build on Windows any more due to error:
```
ConanException: Cannot recognize the Windows subsystem, install MSYS2/cygwin or specify a build_require to apply it.
```

### Steps to reproduce (Include if Applicable)
Local `profile.txt`:
```
include(default)

[settings]
compiler=Visual Studio
compiler.version=14
arch=x86
arch_build=x86
```
`conanfile.txt`:
```
[requires]
openssl/1.0.2t@

[options]
openssl:shared=True
openssl:no_zlib=True

```

### Logs (Include/Attach if Applicable)

```
conan install . --build=missing -p profile.txt
...
Configuration:
[settings]
arch=x86
arch_build=x86
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=14
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

openssl/1.0.2t: WARN: Package is corrupted, removing folder: C:\Users\Vincas\.conan\data\openssl\1.0.2t\_\_\package\e30ae78fe1503c6102f7c756d918ad4e5845a76a
conanfile.txt: Installing package
Requirements
    openssl/1.0.2t from 'conan-center' - Cache
Packages
    openssl/1.0.2t:e30ae78fe1503c6102f7c756d918ad4e5845a76a - Build
Build requirements
    nasm/2.14 from 'conan-center' - Cache
Build requirements packages
    nasm/2.14:b2fcbc60b3bee75023a4f7934cadb723eec550bd - Cache

nasm/2.14: Already installed!
nasm/2.14: Appending PATH environment variable: C:\Users\Vincas\.conan\data\nasm\2.14\_\_\package\b2fcbc60b3bee75023a4f7934cadb723eec550bd\bin
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
openssl/1.0.2t: Applying build-requirement: nasm/2.14
openssl/1.0.2t: WARN: Build folder is dirty, removing it: C:\Users\Vincas\.conan\data\openssl\1.0.2t\_\_\build\e30ae78fe1503c6102f7c756d918ad4e5845a76a
openssl/1.0.2t: Copying sources to build folder
openssl/1.0.2t: Building your package in C:\Users\Vincas\.conan\data\openssl\1.0.2t\_\_\build\e30ae78fe1503c6102f7c756d918ad4e5845a76a
openssl/1.0.2t: Generator txt created conanbuildinfo.txt
openssl/1.0.2t: Calling build()
Conan:vcvars already set 
openssl/1.0.2t: activated option: no_zlib
openssl/1.0.2t: activated option: shared
openssl/1.0.2t: activated option: no_zlib
openssl/1.0.2t: activated option: shared
openssl/1.0.2t: ['"VC-WIN32"', 'shared', '--prefix=/c/users/vincas/.conan/data/openssl/1.0.2t/_/_/package/e30ae78fe1503c6102f7c756d918ad4e5845a76a', '--openssldir=/c/users/vincas/.conan/data/openssl/1.0.2t/_/_/package/e30ae78fe1503c6102f7c756d918ad4e5845a76a/res', 'no-unit-test', 'no-tests', 'no-zlib', 'shared']
openssl/1.0.2t: 
openssl/1.0.2t: ERROR: Package 'e30ae78fe1503c6102f7c756d918ad4e5845a76a' build failed
openssl/1.0.2t: WARN: Build folder C:\Users\Vincas\.conan\data\openssl\1.0.2t\_\_\build\e30ae78fe1503c6102f7c756d918ad4e5845a76a
ERROR: openssl/1.0.2t: Error in build() method, line 499
	self._make()
while calling '_make', line 449
	self.run('{perl} ./Configure {args}'.format(perl=self._perl, args=args), win_bash=self._win_bash)
	ConanException: Cannot recognize the Windows subsystem, install MSYS2/cygwin or specify a build_require to apply it.

```

**Comment by Talkless** [comment]
Why it require MSYS stuff with "native" VS build? Old (non cci) recipe worked fine.

**Comment by Talkless** [comment]
Adding build_requires;
```
[build_requires]
openssl*:msys2/20161025@
```
Gives me this:
```
7zip/19.00: Copying sources to build folder
7zip/19.00: Building your package in C:\Users\Vincas\.conan\data\7zip\19.00\_\_\build\b2fcbc60b3bee75023a4f7934cadb723eec550bd
7zip/19.00: Generator txt created conanbuildinfo.txt
7zip/19.00: Calling build()
Found Visual Studio 15
7zip/19.00:
7zip/19.00: ERROR: Package 'b2fcbc60b3bee75023a4f7934cadb723eec550bd' build failed
7zip/19.00: WARN: Build folder C:\Users\Vincas\.conan\data\7zip\19.00\_\_\build\b2fcbc60b3bee75023a4f7934cadb723eec550bd
ERROR: 7zip/19.00: Error in build() method, line 40
        vcvars = tools.vcvars_command(self.settings)
        ConanException: Error, Visual environment already set to 14
 Current settings visual version: 15
```
Ugh :) .

**Comment by SSE4** [comment]
I think it's the consequence of https://github.com/conan-io/conan-center-index/commit/a0f59615d962d600419e0e0a0da86f30e4b8048c#diff-07f709ffc8e00c0770f75389941696cd
we shouldn't consider x86_64 -> x86 as cross compilation
but in general, condition should be changed (e.g. to allow ARM builds with Visual Studio)
I'll take a look

**Comment by Talkless** [comment]
> I'll take a look

Thanks @SSE4 !

**Comment by madebr** [comment]
@Talkless Are you executing conan in a Visual Studio development console?
Is the result different if you execute it in a plain `cmd` console?

**Comment by Talkless** [comment]
@madebr this was done in the plain cmd. Well, catually, windows power shell -> cmd, as shift + rmb now suggest only power shell...

**Comment by SSE4** [comment]
fixed by #129
